src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md

```diff
@@ -0,0 +1,266 @@
+5. Create a RADOS scrub pool with replication factor 3
+6. Temporarily disable recovery and backfill ops.
+7. Bring up the downed OSD from step 4
+8. Prefill the scrub pool with 100000 objects of 4 MiB size
+9. Initiate deep scrub on the scrub pool and enable recovery and backfill at the same time
+10. Initiate I/O on the client pool for 9 minutes using fio
+11. While the client I/O and deep scrub are running, the test collects statistics related to client throughput and latency. The test also collects scrub statistics such as number of objects scrubbed and scrub duration and recovery statistics.
+
+### Test Results
+
+#### Client Throughput Comparison
+
+![](images/image16_IOPS_mClock_vs_WPQ_recovery_test.png)
+
+The graph above displays the drop in client throughput with recovery and scrub happening in parallel. All the profiles have a similar drop in throughput and the difference between them is not significant. To get a better understanding, we will compare the client latencies in the next sections.
+
+#### Client Latency Comparison
+
+The graph below describes how average client latency varies for the various profiles over the duration of the test.
+
+![](images/image17_avg_clat_while_scrubbing_and_recovery.png)
+
+There is a jump in client latency for the mClock profiles in the beginning of the test. This can be attributed to the beginning of recovery. As we will see in the next section, recovery rate is much higher for mClock profiles as compared to WPQ.
+Over time this spike flattens out and we can see client latency returning to a normal trend and at times even lower than WPQ.
+
+#### Scrub and Recovery Comparison
+
+The graphs below show the scrub and recovery rate over the duration of the test. In the scrub rate comparison, mClock profiles take some time to start scrub whereas with WPQ we can see that scrub starts almost immediately.
+
+![](images/image18_scrub_rate_during_recovery_WPQ_vs_mClock.png)
+
+Almost the opposite trend can be observed in the graph comparing recovery rates. The number of objects recovered starts decreasing rapidly for all the mClock profiles. WPQ seems to have a slow start and eventually takes much longer to complete recovery than the mClock profiles.
+
+![](images/image19_recovery_rate_during_scrub_WPQ_vs_mClock.png)
+
+## Conclusion
+
+To summarize, after taking a look at how the default mClock profiles handle background operations such as scrubbing and recovery, we can say that mClock profiles attempt to finish recovery faster than a scrub.
+In addition to this, the experiments with the mClock custom profile proved that with higher allocations we can achieve a faster scrub. Other than mClock parameters, we also explored the existing Ceph configuration parameters that can be tuned to give us better scrub performance with mClock.
+The mClock scheduler is the default osd_op_queue in the Quincy release. Let us know how your experience with mClock goes and thank you for reading!
```

Comment:
```suggestion
The mClock scheduler is the default osd_op_queue in the Quincy release. Let us know how your experience with mClock goes, and thank you for reading!
```

You can also include a sentence about reaching out to <ceph-users@ceph.io> if anyone has questions.